find_max.rb

```diff
@@ -0,0 +1,19 @@
+
+# Find the maximum element between index 0 and max_index, exclusive
+
+def find_max(array, max_index)
+  max = nil
+  if max_index == 0
+    puts "Array is empty"
+  else
```

Comment:
It might make the code a little less nested to have the emptiness check as its own stand-alone function with a short-circuit return and then run the code currently in the "else" portion of the code independently.